contractcourt/chain_watcher.go

```diff
@@ -30,20 +30,78 @@ const (
 	maxCommitPointPollTimeout = 10 * time.Minute
 )
 
-// LocalUnilateralCloseInfo encapsulates all the informnation we need to act
-// on a local force close that gets confirmed.
+// LocalUnilateralCloseInfo encapsulates all the informnation we need to act on
+// a local force close that gets confirmed.
 type LocalUnilateralCloseInfo struct {
 	*chainntnfs.SpendDetail
 	*lnwallet.LocalForceCloseSummary
 	*channeldb.ChannelCloseSummary
+
+	// CommitSet is the set of known valid commitments at the time the
+	// remote party's commitemnt hit the chain.
+	CommitSet CommitSet
 }
 
-// CooperativeCloseInfo encapsulates all the informnation we need to act
-// on a cooperative close that gets confirmed.
+// CooperativeCloseInfo encapsulates all the informnation we need to act on a
+// cooperative close that gets confirmed.
 type CooperativeCloseInfo struct {
 	*channeldb.ChannelCloseSummary
 }
 
+// RemoteUnilateralCloseInfo wraps the normal UnilateralCloseSummary to couple
+// the CommitSet at the time of channel closure.
+type RemoteUnilateralCloseInfo struct {
+	*lnwallet.UnilateralCloseSummary
+
+	// CommitSet is the set of known valid commitments at the time the
+	// remote party's commitemnt hit the chain.
+	CommitSet CommitSet
+}
+
+// CommitSet is a collection of the set of known validate commitments at a
+// given instant. If ConfCommitKey is set, then the commitment identified by
+// the HtlcSetKey has hit the chain. This struct will be used to examine all
+// live HTLCs to determine if any additional actions need to be made based on
+// the remote party's commitments.
+type CommitSet struct {
+	// ConfCommitKey if non nil, then the commitment identified by this
+	// HtlcSetKey has hit the chain.
+	ConfCommitKey *HtlcSetKey
+
+	// LocalCommit is the latest valid commitment for the local party.
+	LocalCommit channeldb.ChannelCommitment
+
+	// RemoteCommit is the latest valid commitment for the remote party.
+	RemoteCommit channeldb.ChannelCommitment
+
+	// RemotePendingCommit if non nil, is the pending commmitment for the
+	// remote party. This is a commitment that we've sent to them, but have
+	// not yet received a revocation for.
+	RemotePendingCommit *channeldb.ChannelCommitment
+}
+
+// toActiveHTLCSets returns the set of all active HTLCs across all commitment
+// transactions.
+func (c *CommitSet) toActiveHTLCSets() map[HtlcSetKey]htlcSet {
+	htlcSets := make(map[HtlcSetKey]htlcSet)
+
+	htlcSets[HtlcSetKey{
+		IsRemote: false, IsPending: false,
```

Comment:
would help readability to use the defined keys, e.g. `LocalHtlcSetKey`

Reply:
Fixed, forgot to update this after I aded the definitions.